src/drivers/video/video_api.h

```diff
@@ -0,0 +1,110 @@
+#ifndef _VIDEO_API_H_
+#define _VIDEO_API_H_
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define MAKE_RGBA(r, g, b, a) ((a << 24) | (r << 16) | (g << 8) | (b))
+#define MAKE_RGB(r, g, b)     MAKE_RGBA(r, g, b, 0xFF)
+
+typedef struct video_driver
+{
+  uint32_t* buffer;
+  void* native_buffer;
+  uint32_t width;
+  uint32_t height;
+  void* pimpl;
+  void (*swap_buffers)();
+} video_driver_t;
+
+extern video_driver_t video_driver;
+
+/**
+ * Clears the screen with a specific color
+ *
+ * @param color the color to clear with. You can use MAKE_RGB/RGBA to make one.
+ */
+void video_clear(uint32_t color);
+
+/**
+ * Clears a region of the screen with a specific color
+ *
+ * @param color the color to clear with. You can use MAKE_RGB/RGBA to make one.
+ * @param x the x coordinate of the rectangle you want to clear
+ * @param y the y coordinate of the rectangle you want to clear
+ * @param width the width of the rectangle you want to clear
+ * @param height the height of the rectangle you want to clear
+ */
+void video_clear_region(uint32_t color,
+                        int32_t x,
+                        int32_t y,
+                        int32_t width,
+                        int32_t height);
+
+/**
+ * Gets the width of the video screen
+ */
+static inline uint32_t video_width()
+{
+  return video_driver.width;
```

Comment:
can we move code in `.c` files?

Reply:
Since it's inline we can't, and it saves us a lot of performance when accessing this function